docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,319 @@
+ */
+HexString: 'hex' (('"' EvenHexDigits? '"') | ('\'' EvenHexDigits? '\''));
+/**
+ * Hex numbers consist of a prefix and an arbitrary number of hex digits that may be delimited by underscores.
+ */
+HexNumber: '0' [xX] HexDigits;
+//@doc:inline
+fragment HexDigits: HexCharacter ('_'? HexCharacter)*;
+//@doc:inline
+fragment EvenHexDigits: HexCharacter HexCharacter ('_'? HexCharacter HexCharacter)*;
+//@doc:inline
+fragment HexCharacter: [0-9A-Fa-f];
+
+/**
+ * A decimal number literal consists of decimal digits that may be delimited by underscores and
+ * an optional positive or negative exponent.
+ * If the digits contain a decimal point, the literal has fixed point type.
+ */
+DecimalNumber: (DecimalDigits | (DecimalDigits? '.' DecimalDigits)) ([eE] '-'? DecimalDigits)?;
+//@doc:inline
+fragment DecimalDigits: [0-9] ('_'? [0-9])* ;
+
+
+/**
+ * An identifier in solidity has to start with a letter, a dollar-sign or an underscore and
+ * may additionally contain numbers after the first symbol.
+ */
+Identifier: IdentifierStart IdentifierPart*;
+//@doc:inline
+fragment IdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment IdentifierPart: [a-zA-Z0-9$_];
+
+WS: [ \t\r\n\u000C]+ -> skip ;
+COMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+LINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN);
+
+mode AssemblyBlockMode;
+
+//@doc:inline
+AssemblyDialect: '"evmasm"';
+AssemblyLBrace: '{' -> popMode, pushMode(YulMode);
+
+AssemblyBlockWS: [ \t\r\n\u000C]+ -> skip ;
+AssemblyBlockCOMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+AssemblyBlockLINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN) ;
+
+mode YulMode;
+
+YulBreak: 'break';
+YulCase: 'case';
+YulContinue: 'continue';
+YulDefault: 'default';
+YulFalse: 'false';
+YulFor: 'for';
+YulFunction: 'function';
+YulIf: 'if';
+YulLeave: 'leave';
+YulLet: 'let';
+YulSwitch: 'switch';
+YulTrue: 'true';
+
+/**
+ * Builtin functions in the EVM Yul dialect.
+ */
+YulEVMBuiltin:
+    'stop' | 'add' | 'sub' | 'mul' | 'div' | 'sdiv' | 'mod' | 'smod' | 'exp' | 'not'
+    | 'lt' | 'gt' | 'slt' | 'sgt' | 'eq' | 'iszero' | 'and' | 'or' | 'xor' | 'byte'
+    | 'shl' | 'shr' | 'sar' | 'addmod' | 'mulmod' | 'signextend' | 'keccak256'
+    | 'pop' | 'mload' | 'mstore' | 'mstore8' | 'sload' | 'sstore' | 'msize' | 'gas'
+    | 'address' | 'balance' | 'selfbalance' | 'caller' | 'callvalue' | 'calldataload'
+    | 'calldatasize' | 'calldatacopy' | 'extcodesize' | 'extcodecopy' | 'returndatasize'
+    | 'returndatacopy' | 'extcodehash' | 'create' | 'create2' | 'call' | 'callcode'
+    | 'delegatecall' | 'staticcall' | 'return' | 'revert' | 'selfdestruct' | 'invalid'
+    | 'log0' | 'log1' | 'log2' | 'log3' | 'log4' | 'chainid' | 'origin' | 'gasprice'
+    | 'blockhash' | 'coinbase' | 'timestamp' | 'number' | 'difficulty' | 'gaslimit';
+
+YulLBrace: '{' -> pushMode(YulMode);
+YulRBrace: '}' -> popMode;
+YulLParen: '(';
+YulRParen: ')';
+YulAssign: ':=';
```

Comment:
Well, this is an obsolete keyword, wouldn't call it YulAssign (it is called AssemblyAssign), but the consistency of having `Yul` as a perfix is nice. Perhaps leave a comment this is not currently used?

Reply:
What are you talking about?

Reply:
Damn, mistook this for `=:`, which is the stack assign.

Reply:
That's just the plain regular Yul assignment. The fact that it can't occur at the start of a statement has nothing to do with the lexer or the fact that it's a keyword...